lang/src/org/partiql/lang/syntax/PartiQLParser.kt

```diff
@@ -59,26 +61,65 @@ internal class PartiQLParser(
         return visitor.visit(tree) as PartiqlAst.Statement
     }
 
-    private fun parseQuery(lexer: Lexer): ParseTree {
-        val parser = getParser(lexer)
+    /**
+     * To reduce latency costs, the [PartiQLParser] attempts to use [PredictionMode.SLL] and falls back to
```

Comment:
Perhaps: "As performance optimization, first attempt to parse using ANTLR's fast, but less powerfull [PredictionMode.SLL], falling back to the slower [PredictionMode.LL] that is capable of parsing all valid ANTLR grammars."